aidb/engine/approx_select_engine.py

```diff
@@ -0,0 +1,224 @@
+          recall_target,
+          confidence,
+          total_length
+      )
+
+      R1 = sorted_satisfied_sampled_results.index
+      R2 = dataset[dataset[PROXY_SCORE] >= tau_modified].index
+
+      additional_samples = list(set(R1).union(set(R2)))
+
+      print('num_samples', len(additional_samples))
+      logging.info(f'num_samples: {len(additional_samples)}')
+
+      additional_satisfied_sampled_results,  additional_all_sampled_results = await self.get_inference_results(
+          query,
+          additional_samples,
+          conn
+      )
+
+    return additional_satisfied_sampled_results
+
+
+  async def get_inference_results(self, query:Query, sampled_index: List[int], conn):
+    # TODO: rewrite query, use full scan to execute query
+    bound_service_list = query.inference_engines_required_for_query
+    for bound_service in bound_service_list:
+      inp_query_str = self.get_input_query_for_inference_service_filtered_index(bound_service,
+                                                                                self.blob_mapping_table_name,
+                                                                                sampled_index)
+      inp_df = await conn.run_sync(lambda conn: pd.read_sql(text(inp_query_str), conn))
+      inp_df.set_index(VECTOR_ID_COLUMN, inplace=True, drop=True)
+      await bound_service.infer(inp_df)
+
+
+    no_aqp_query = Query(query.base_sql_no_aqp.sql(), self._config)
+    new_exp = no_aqp_query.add_select(
+        no_aqp_query.expression_after_normalize_columns,
+        f'{self.blob_mapping_table_name}.{VECTOR_ID_COLUMN}'
+    )
+
+    blob_keys = [col.name for col in self._config.tables[self.blob_mapping_table_name].columns]
+    added_cols = set()
+    join_conditions = []
+    tables_in_query = no_aqp_query.tables_in_query
+    table_alias = no_aqp_query.table_name_to_aliases
+    for table_name in tables_in_query:
+      for col in self._config.tables[table_name].columns:
+        col_name = col.name
+        if col_name in blob_keys and col_name not in added_cols:
+          if table_name in table_alias:
+            table_name = table_alias[table_name]
+          join_conditions.append(f'{self.blob_mapping_table_name}.{col_name} = {table_name}.{col_name}')
+          added_cols.add(col_name)
+    join_conditions_str = ' AND '.join(join_conditions)
+
+    new_exp = no_aqp_query.add_join(new_exp, f'JOIN {self.blob_mapping_table_name} ON {join_conditions_str}')
+    new_query = Query(new_exp.sql(), self._config)
+
+    table_columns = [f'{self.blob_mapping_table_name}.{VECTOR_ID_COLUMN}']
+    sampled_df = pd.DataFrame({VECTOR_ID_COLUMN: sampled_index})
+
+    all_query_exp, _ = self.add_filter_key_into_query(
+        table_columns,
+        sampled_df,
+        new_query,
+        new_query.base_sql_no_where
+    )
+
+    all_df = await conn.run_sync(lambda conn: pd.read_sql(text(all_query_exp.sql()), conn))
+    # drop duplicated columns, this will happen when 'select *'
```

Comment:
Why does select * lead to duplicated columns?

Reply:
because we need to add '__vector_id' in SELECT clause. When SELECT *, there will be two '__vector_id' columns.

Reply:
Please mention it in the comment itself

Reply:
@ttt-77 please resolve this